proxy-node-shared/src/main/java/uk/gov/ida/notification/exceptions/mappers/BaseExceptionMapper.java

```diff
@@ -18,53 +19,54 @@
     private static final Logger LOG = LoggerFactory.getLogger(BaseExceptionMapper.class);
 
     private UriInfo uriInfo;
-    private HttpServletRequest httpServletRequest;
-
     private String logId;
-    private String authnRequestId;
-    private String issuerId;
-    private DateTime issueInstant;
 
     @Context
     public void setUriInfo(UriInfo uriInfo) {
         this.uriInfo = uriInfo;
     }
 
-    @Context
-    public void setHttpServletRequest(HttpServletRequest httpServletRequest) {
-        this.httpServletRequest = httpServletRequest;
-    }
-
     @Override
     public Response toResponse(TException exception) {
         this.logId = String.format("%016x", ThreadLocalRandom.current().nextLong());
 
-        handleException(exception);
         logException(exception);
 
-        return getResponse(exception);
+        return Response.status(getResponseStatus())
+                .type(MediaType.APPLICATION_JSON_TYPE)
+                .entity(new ErrorMessage(
+                        getResponseStatus().getStatusCode(),
+                        getResponseMessage(exception)))
+                .build();
     }
 
-    protected abstract void handleException(TException exception);
+    protected abstract Response.Status getResponseStatus();
 
-    protected abstract Response getResponse(TException exception);
+    protected abstract String getResponseMessage(TException exception);
 
     String getLogId() {
         return logId;
     }
 
-    void setAuthnRequestValues(String authnRequestId, String issuerId, DateTime issueInstant) {
-        this.authnRequestId = authnRequestId;
-        this.issuerId = issuerId;
-        this.issueInstant = issueInstant;
+    // TODO: Log these three properties via the logger itself as context attributes
+    protected String getAuthnRequestId(TException exception) {
+        return null;
+    }
+
+    protected String getIssuerId(TException exception) {
+        return null;
+    }
+
+    protected DateTime getIssueInstant(TException exception) {
+        return null;
     }
 
     private void logException(TException exception) {
         String message = exception.getMessage();
         String logId = String.format("%016x", ThreadLocalRandom.current().nextLong());
 
         LOG.warn(format("Error whilst contacting uri [{0}]; logId: {1}; requestId: {2}; issuer: {3}; issueInstant: {4}; cause: {5}",
```

Comment:
Could BaseExceptionMapper have a `Level getLogLevel()` method that returns `warn` but subclasses can override? In that way a subclass could indicate that it wants to log at level warn or info etc.

Reply:
I think they should all log at the `error` level tbh. Since these are error that will result in a failure response being sent back to the connector node.